tests/app/main/test_frameworks.py

```diff
@@ -2805,14 +2810,188 @@ def test_get_with_with_previous_answers(self, data_api_client):
             doc = html.fromstring(res.get_data(as_text=True))
             assert len(doc.xpath('//input[@id="input-PR1-no"]/@checked')) == 1
 
-    def test_post_valid_data(self, data_api_client):
+    def test_get_with_with_prefilled_answers(self, data_api_client):
         with self.app.test_client():
             self.login()
+            # First call is for the current framework; second call for the framework to pre-fill from
+            data_api_client.get_framework.side_effect = [
+                self.framework(slug='g-cloud-9', name='G-Cloud 9', status='open'),
+                self.framework(slug='digital-outcomes-and-specialists-2', name='Digital Stuff 2', status='live'),
+            ]
+            # Current framework application information
+            data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+                framework_slug="g-cloud-9",
+                declaration={"status": "started"},
+                prefill_declaration_from_framework_slug="digital-outcomes-and-specialists-2"
+            )
+            # The previous declaration to prefill from
+            data_api_client.get_supplier_declaration.return_value = {
```

Comment:
I don't tend to get too bound up by terms, what's "unit testing" or not "unit testing" - I'm just concerned with what's useful testing (hey maybe I should register _that_ term). When a test fails for me, the first thing I look at isn't the test name (take a survey of test names we have - more often that not they aren't particularly illuminating), I run it under `--pdb` and look at the assertion that failed, which hopefully is written in a way that makes it clear what it's trying to assert for and why.

Yes, testing for the govuk banner is going too far, but this is where you have to apply some pragma. It's also something that should have been tested as part of its own module. Code such as a view should be able to consider something as fundamental as the banner to be a black box that is tested on its own and "works".

And while "tests are cheap", maintaining them certainly is not. How may people when they fix up a particular small test go and similarly fix up the 57 other variants of the test sitting next to it? Copy pasted test variants have a tendency to diverge... 

Test execution time is also something that isn't particularly cheap.

@TheDoubleK I think only when the situation falls into the category of "leave the function some leeway of how it wants to achieve this, but if it's "there's no possible way for it to give the right answer if it doesn't call the right thing", I'd say not. If we want to go mad, I'd say the right thing to do would be for the `get_supplier_declaration` mock to be able to return appropriate mock declarations for either framework depending on how it's called and for the assertions to be able to distinguish between them. But that would be insane to set up manually all the time, which is one reason I'd like to move to a more "comprehensive" api-mocking system.